point_viewer_grpc/src/service.rs

```diff
@@ -266,57 +264,67 @@ impl OctreeService {
-                        v.set_x(p.position.x);
-                        v.set_y(p.position.y);
-                        v.set_z(p.position.z);
-                        reply.mut_positions().push(v);
-                    }
-
-                    {
-                        let mut v = point_viewer::proto::Color::new();
-                        let clr = p.color.to_f32();
-                        v.set_red(clr.red);
-                        v.set_green(clr.green);
-                        v.set_blue(clr.blue);
-                        v.set_alpha(clr.alpha);
-                        reply.mut_colors().push(v);
-                    }
-
-                    if let Some(i) = p.intensity {
-                        reply.mut_intensities().push(i);
-                    }
-
-                    reply_size += bytes_per_point;
-                    if reply_size > max_message_size - bytes_per_point {
-                        tx.send((reply.clone(), WriteFlags::default())).unwrap();
-                        reply.mut_positions().clear();
-                        reply.mut_colors().clear();
-                        reply.mut_intensities().clear();
-                        reply_size = 0;
-                    }
-                };
-                match query {
-                    OctreeQuery::Box(bounding_box) => service_data
-                        .octree
-                        .points_in_box(&bounding_box)
-                        .for_each(func),
-                    OctreeQuery::Frustum(frustum_matrix) => service_data
-                        .octree
-                        .points_in_frustum(&frustum_matrix)
-                        .for_each(func),
-                    OctreeQuery::FullPointcloud => service_data.octree.all_points().for_each(func),
-                    OctreeQuery::OrientedBeam(beam) => service_data
-                        .octree
-                        .points_in_oriented_beam(&beam)
-                        .for_each(func),
+                // this function is currently not efficiently implemented
+                let func = |p_data: PointData| {
+                    reply.positions = p_data
+                        .position
+                        .iter()
+                        .map(|p| {
+                            let mut v = point_viewer::proto::Vector3d::new();
+                            v.set_x(p.x);
+                            v.set_y(p.y);
+                            v.set_z(p.z);
+                            v
+                        })
+                        .collect();
+
+                    reply.colors = match p_data.layers.get(&"color".to_string()) {
+                        Some(LayerData::U8Vec4(data)) => data
+                            .iter()
+                            .map(|p| {
+                                let mut v = point_viewer::proto::Color::new();
+                                v.set_red(f32::from(p.x) / 255.);
```

Comment:
How about using `Color::to_f32` to perform the conversion?

Reply:
I changed it, it adds some lines but the performance seems not to be affected from that